src/argilla/client/feedback/dataset/remote.py

```diff
@@ -0,0 +1,257 @@
+
+    def __repr__(self) -> str:
+        return (
+            "The `records` of a `FeedbackDataset` in Argilla are being lazily"
+            " fetched, and never stored locally. You can either loop over `records`"
+            " or access them by index, and those will be fetched from Argilla on the"
+            " fly."
+        )
+
+    def __parse_record(self, record: "FeedbackItemModel") -> RemoteFeedbackRecord:
+        record = record.dict(
+            exclude={
+                "inserted_at": ...,
+                "updated_at": ...,
+                "responses": {"__all__": {"id", "inserted_at", "updated_at"}},
+                "suggestions": {"__all__": {"id"}},
+            },
+            exclude_none=True,
+        )
+        for suggestion in record.get("suggestions", []):
+            suggestion.update({"question_name": self.__question_id2name[suggestion["question_id"]]})
+        return RemoteFeedbackRecord(client=self.client, name2id=self.__question_name2id, **record)
+
+    def __len__(self) -> int:
+        try:
+            response = datasets_api_v1.get_metrics(client=self.client, id=self.id)
+        except Exception as e:
+            raise Exception(
+                f"Failed while getting the metrics from the current `FeedbackDataset` in Argilla with exception: {e}"
+            ) from e
+        return response.parsed.records.count
+
+    def __getitem__(self, key: Union[slice, int]) -> Union[RemoteFeedbackRecord, List[RemoteFeedbackRecord]]:
+        offsets = []
+        limit = None
+        if isinstance(key, slice):
+            start, stop, step = key.indices(len(self))
+            if step is not None and step != 1:
+                return [self[i] for i in range(start, stop, step)]
+            if start < 0:
+                start += len(self)
+            if stop < 0:
+                stop += len(self)
+            if start < 0 or stop < 0:
+                raise IndexError("Index out of range")
+            limit = stop - start
+            offsets = [start] if limit < FETCHING_BATCH_SIZE else list(range(start, stop, FETCHING_BATCH_SIZE))
+        elif isinstance(key, int):
+            if key < 0:
+                key += len(self)
+            if key < 0 or key >= len(self):
+                raise IndexError("Index out of range")
+            offsets = [key]
+            limit = 1
+        else:
+            raise TypeError("Invalid argument type")
+
+        records = []
+        for offset in offsets:
+            fetched_records = datasets_api_v1.get_records(
+                client=self.client,
+                id=self.id,
+                offset=offset,
+                limit=limit,
+            ).parsed
+            if len(fetched_records.items) == 1:
+                record = fetched_records.items[0]
+                records.append(self.__parse_record(record))
+            else:
+                records.extend([self.__parse_record(record) for record in fetched_records.items])
```

Comment:
Why we cannot do something like:
```suggestion
            records.extend([self.__parse_record(record) for record in fetched_records.items])
```

Reply:
Thanks makes sense!